src/system_identity.rs

```diff
@@ -1 +1,83 @@
+pub mod creation_response;
+pub mod generator;
+pub mod identity_creator;
 pub mod input_data;
+
+/// System identity information. Final output of the System Identity creation process.
+#[derive(Debug, Clone)]
+pub struct SystemIdentity {
+    pub id: String,
+    pub name: Option<String>,
+    pub client_id: String,
+    pub organization_id: String,
+    pub identity_type: SystemIdentityType,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum SystemIdentityType {
+    L1 {
+        client_secret: ClientSecret,
+        credential_expiration: String,
+    },
+    L2 {
+        pub_key: Base64PublicKey,
+    },
+}
+
+type ClientSecret = String; // For L1 System Identity. Type better? What are we doing with these?
+type Base64PublicKey = String; // For L2 System Identity. Type better? What are we doing with these?
```

Comment:
I don't think they are wrongly typed as string. Regarding how they are used, I think that's responsibility for the client of the Generator, so the generator should not care :) 
TL;DR: I think these are ok as they are (at least for now)

Reply:
Removed the comments about typing in https://github.com/newrelic/newrelic-auth-rs/pull/33/commits/c4438527a577559010a036e0d05f220962e4d10d 👌